jfix-stdlib-concurrency/src/main/java/ru/fix/stdlib/concurrency/futures/PendingFutureLimiter.java

```diff
@@ -206,7 +208,9 @@ public PendingFutureLimiter changeThresholdFactor(float thresholdFactor) {
             }
             return null;
         });
-        return future;
+        counter.incrementAndGet();
+        pendingFutures.put(resultFuture, System.currentTimeMillis());
```

Comment:
releaseTimeoutedIfPossible

pendingFutures.entrySet().stream().filter(isTimeoutPredicate).forEach(completeExceptionally);
if (pendingFutures.entrySet().stream().anyMatch(isTimeoutPredicate)) {
    log.error(errorMessage);
}
after first line futures will be removed from map so in second line we will have nothing to print.  Write error mesage within forEach consumer

Reply:
It is better to completely remove logging of this message from this method. It will be logged inside this future's .handleAsync(..). This way it also will be printed exactly once for a future.

Reply:
I can also try to collect one nice message for all dequeued batch of out dated futures, but I'll have to control it's consistency, knowing that releaseTimeoutedIfPossible can be called in parallel. 
This way, I'll also remove all these single messages logging from the future's handleAsync. 

So this solution doesn't seems to be neither simple nor necessary for me. But it is not impossible neither.